types/minio/index.d.ts

```diff
@@ -429,6 +484,57 @@ export class CloudFunctionConfig extends TargetConfig {
     constructor(arn: string);
 }
 
+export interface CopySourceOptionsHeaderOptions {
+    "x-amz-copy-source": string;
+    "x-amz-copy-source-if-match"?: string;
+    "x-amz-copy-source-if-none-match"?: string;
+    "x-amz-copy-source-if-modified-since"?: string;
+    "x-amz-copy-source-if-unmodified-since"?: string;
+}
+
+export class CopySourceOptions {
+    constructor(
+        Bucket: string,
+        Object: string,
+        VersionID?: string,
+        MatchETag?: string,
+        NoMatchETag?: string,
+        MatchModifiedSince?: string,
+        MatchUnmodifiedSince?: string,
+        MatchRange?: boolean,
+        Start?: number,
+        End?: number,
+        Encryption?: {
+            type: string;
+            SSEAlgorithm?: string;
+            KMSMasterKeyID?: string;
+        },
+    );
+
+    getHeaders(): CopySourceOptionsHeaderOptions;
+    validate(): boolean;
+}
+
+export class CopyDestinationOptions {
+    constructor(
+        Bucket: string,
+        Object: string,
+        Encryption?: {
+            type: string;
+            SSEAlgorithm?: string;
+            KMSMasterKeyID?: string;
+        },
+        UserMetadata?: EmptyObject,
+        UserTags?: EmptyObject | string,
```

Comment:
Here the `EmptyObject`s mean `{}` literally so `Record<string, unknown>` would fit better

Reply:
Fixed!